### PR TITLE
Redact sensitive user information from log

### PR DESCRIPTION
**Pull request by tareq89**
https://github.com/opencrvs/opencrvs-core/issues/7181

### CODE REVIEW

**Comment by tareq89** [comment]
Reason for removing following headers from the system log:
req.headers.authorization: This header often contains sensitive authentication tokens or credentials (e.g., Bearer tokens). If leaked, these tokens can be used by malicious actors to gain unauthorized access to the system. Therefore, omitting this field helps to protect user accounts and sensitive data.

req.remoteAddress: This property contains the IP address of the client making the request. Exposing IP addresses can reveal the geographical location and identity of users, making them vulnerable to targeted attacks and privacy breaches.

req.headers['x-real-ip']: Similar to req.remoteAddress, this header can contain the real IP address of the client. Omitting it helps in protecting user privacy by not exposing their actual IP address.

req.headers['user-agent']: The User-Agent header provides detailed information about the client’s device, operating system, and browser version. While not as sensitive as some other fields, it can still be used to fingerprint and track users. Removing it helps in reducing the potential for user profiling.

req.headers['x-real-user-agent']: This is another form of the User-Agent header, potentially revealing the same detailed client information. Omitting it helps to further protect user privacy by preventing detailed client environment disclosure.

req.headers['traceparent']: This header is used for distributed tracing, and can contain sensitive information about the request path through various services. If leaked, it can provide insights into the system architecture and request flow, which can be exploited by attackers.

req.headers['tracestate']: Similar to traceparent, this header is part of distributed tracing and can include information that helps in correlating tracing information across services. Omitting it helps in protecting the internal tracing data.

req.headers['elastic-apm-traceparent']: This is another header used for tracing by Elastic APM (Application Performance Monitoring). It can contain sensitive data regarding the request's path and interactions. Removing it helps in securing tracing details.

req.headers['x-correlation-id']: The Correlation ID is used to trace and correlate requests across different services and components. If exposed, it can be used by attackers to trace the flow of a specific request through the system, potentially uncovering internal processes and vulnerabilities.

**Comment by tareq89** [comment]
### Reason for removing following headers from the system log:

`req.headers.authorization`: This header often contains sensitive authentication tokens or credentials (e.g., Bearer tokens). If leaked, these tokens can be used by malicious actors to gain unauthorized access to the system. Therefore, omitting this field helps to protect user accounts and sensitive data.

`req.remoteAddress`: This property contains the IP address of the client making the request. Exposing IP addresses can reveal the geographical location and identity of users, making them vulnerable to targeted attacks and privacy breaches.

`req.headers['x-real-ip']`: Similar to req.remoteAddress, this header can contain the real IP address of the client. Omitting it helps in protecting user privacy by not exposing their actual IP address.

`req.headers['user-agent']`: The User-Agent header provides detailed information about the client’s device, operating system, and browser version. While not as sensitive as some other fields, it can still be used to fingerprint and track users. Removing it helps in reducing the potential for user profiling.

`req.headers['x-real-user-agent']`: This is another form of the User-Agent header, potentially revealing the same detailed client information. Omitting it helps to further protect user privacy by preventing detailed client environment disclosure.

`req.headers['traceparent']`: This header is used for distributed tracing, and can contain sensitive information about the request path through various services. If leaked, it can provide insights into the system architecture and request flow, which can be exploited by attackers.

`req.headers['tracestate']`: Similar to traceparent, this header is part of distributed tracing and can include information that helps in correlating tracing information across services. Omitting it helps in protecting the internal tracing data.

`req.headers['elastic-apm-traceparent']`: This is another header used for tracing by Elastic APM (Application Performance Monitoring). It can contain sensitive data regarding the request's path and interactions. Removing it helps in securing tracing details.

`req.headers['x-correlation-id']`: The Correlation ID is used to trace and correlate requests across different services and components. If exposed, it can be used by attackers to trace the flow of a specific request through the system, potentially uncovering internal processes and vulnerabilities.

File: packages/commons/src/logger.ts [hunk 1]
```diff
@@ -11,7 +11,19 @@
...
+    ? pino({
+        redact: [
+          'req.headers.authorization',
+          'req.remoteAddress',
+          "req.headers['x-real-ip']",
+          "req.headers['user-agent']",
+          "req.headers['x-real-user-agent']",
+          "req.headers['traceparent']",
+          "req.headers['tracestate']",
+          "req.headers['elastic-apm-traceparent']",
```
Comment (Zangetsu101) [inline packages/commons/src/logger.ts:23]: I'm not entirely sure but aren't these 3 used by kibana for tracing a request? So if we remove them would it still work?

Comment (rikukissa) [inline packages/commons/src/logger.ts:23]:
```suggestion
```

Good point. I don't think user-agent should be removed either. There's a balance between being able to debug and get information from the logs and not exposing anything that can endanger the user's security / privacy.

**Review by rikukissa — APPROVED** [review]

**Pull request merged**